File: 2015/day1.py
# part 1
# with open("2015/day1input.txt", 'r') as f:
#     stringy = f.readlines()[0].strip()
#     res = 0
#     for c in stringy:
#         if c == '(':
#             res += 1
#         else: #')'
#             res -= 1
#     print(res)

# part 2
with open("2015/day1input.txt", 'r') as f:
    stringy = f.readlines()[0].strip()
    res = 0
    index = 1
    for c in stringy:
        if c == '(':
            res += 1
        else: #')'
            res -= 1
        if res < 0:
            break
        else:
            index += 1
    print(index)

File: 2015/day1.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;
int main(){
    ifstream f;
    string paras;
    f.open("day1input.txt");
    f >> paras;
    f.close();

    // part 1
    int res = 0;
    for(char const &c : paras){
        if (c == '('){
            res++;
        } else {
            res--;
        }
    }
    cout << res << "\n";

    // part 2
    res = 0;
    int i = 0;
    for(char const &c : paras){
        if (c == '('){
            res++;
        } else {
            res--;
        }
        i++;
        if(res < 0){
            cout << i;
            break;
        }
    }
}